manifests/enterprise/config.pp

```diff
@@ -0,0 +1,149 @@
+# Class splunk::enterprise::config
+#
+class splunk::enterprise::config(
+){
+
+  if $splunk::enterprise::manage_password {
+    file { $splunk::enterprise::password_config_file:
+      ensure  => file,
+      owner   => $splunk::enterprise::splunk_user,
+      group   => $splunk::enterprise::splunk_user,
+      content => $splunk::enterprise::password_content,
+    }
+
+    file { $splunk::enterprise::secret_file:
+      ensure  => file,
+      owner   => $splunk::enterprise::splunk_user,
+      group   => $splunk::enterprise::splunk_user,
+      content => $splunk::enterprise::secret,
+    }
+  }
+
+  # Remove init.d file if the service provider is systemd
+  if $facts['service_provider'] == 'systemd' and versioncmp($splunk::enterprise::version, '7.2.2') >= 0 {
```

Comment:
do we need the `versioncmp()` here? Isn't it enough to check for the `service_provider`?

Reply:
Splunk did not support systemd until 7.2.2, so enabling bootstart will drop a sysvinit file